control/prometheus.py

```diff
@@ -146,13 +157,27 @@ def _get_spdk_thread_stats(self):
     def _get_subsystems(self):
         return rpc.nvmf.nvmf_get_subsystems(self.spdk_rpc_client)
```

Comment:
This might be problematic. We experienced some corruptions when there were two SPDK calls in parallel. This is why we introduced an RPC lock in the code and make sure we lock it before calling SPDK. You can replace the call to SPDK here with a call to list_subsystems() using pb2.list_subsystems_req(). This will go through our gRPC and will take the lock.

Reply:
ACK

Reply:
Hi @gbregman 
I've hit a little problem..the message format returned drops a lot of the content that the straight spdk rpc call returned, so things like allow_any_host, the listen addresses, hosts etc are no longer visible to the exporter code. To get the mostly the same information, I need to execute multiple commands; subsystem list, then listener list -n <nqn> and host list -n <nqn> and this will happen every n seconds - are you ok with that?

Reply:
@pcuzner the latest commit brought back the get_subsystems. So you can now call it via gRPC and get all the information you need.

If you need the max_namespaces, please open an issue and we will add it.